### ISSUE TITLE
SNOW-1468522: ERROR 091032: Your client app version, snowcli 2.4.0, is using a deprecated pre-signed URL for PUT.

### ISSUE THREAD

**Issue by APTy** [snowflakedb/snowflake-cli#1175]
### SnowCLI version

Snowflake CLI version: 2.4.0

### Python version

Python 3.11.6

### Platform

macOS-14.2-arm64-arm-64bit

### What happened

Running `snow app init` and then `snow app run` doesn't seem to work in GCP regions. I've looked into the versions, and I'm on the all the latest as far as I can tell. Could I be missing something?

### Console output

```Shell
Checking if stage exists, or creating a new one if none exists.
Performing a diff between the Snowflake stage and your local deploy_root ('/Users/<...>/output/deploy') directory.
There are no new files that exist only on the stage.
There are no existing files that have been modified, or their status is unknown.
There are no existing files that are identical to the ones on the stage.
New files only on your local:
README.md
manifest.yml
setup_script.sql
Uploading diff-ed files from your local /Users/<...>/output/deploy directory to the Snowflake stage.
2024-06-07 17:16:54 ERROR 091032 (0A000): 01b4dc50-0002-4213-0001-fd4202f270fe: Your client app version, snowcli 2.4.0, is using a deprecated pre-signed URL for PUT. Please upgrade to a version that supports GCP downscoped token. See https://community.snowflake.com/s/article/faq-2023-client-driver-deprecation-for-GCP-customers.
╭─ Error ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                                                                                 │
│                                                                                                                                                 │
│     Could not successfully execute the Snowflake SQL statements.                                                                                │
│                                                                                                                                                 │
│                                                                                                                                                 │
│                                                                                                                                                 │
╰─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

### How to reproduce

Follow the instructions for [Tutorial: Developing a Snowflake Native App with the Snowflake Native App Framework](https://docs.snowflake.com/en/developer-guide/native-apps/tutorials/getting-started-tutorial#add-application-logic-and-install-your-first-app)

Set up connection to a Snowflake account in GCP with a basic manifest.yml and run:

```
snow app run -c <account> -p <app-name-dir>
```

I've even run this from latest pip-installed snowcli in a virtual env to version 2.4.0 with the following lib versions, which should be fine since the compatibility matrix shows we need anything higher than `2.7.8`:
```
snowflake-cli-labs==2.4.0
snowflake-connector-python==3.10.0
snowflake-snowpark-python==1.18.0
snowflake.core==0.8.0
```

**Comment by APTy** [comment]
Update! Just for kicks, I installed `2.2.0` and it worked without any errors! Maybe something got introduced recently?

**Comment by sfc-gh-turbaszek** [comment]
Solved via #1187
